### ISSUE TITLE
TAG review request: WebAssembly Threads

### ISSUE THREAD

**Issue by binji** [w3ctag/design-reviews#299]
Bonjour TAG,

I'm requesting a TAG review of:

  - Name: WebAssembly Threads
  - Specification URL: https://github.com/WebAssembly/threads/blob/master/proposals/threads/Overview.md#webassemblymemory-constructor
    (will move this to bikeshed soon, which is used for the rest of the Wasm JS-API)
  - Explainer, Requirements Doc, or Example code: https://github.com/WebAssembly/threads/blob/master/proposals/threads/Overview.md
  - Tests: None currently
  - Primary contacts: binji@

Further details (optional):

  - Relevant time constraints or deadlines: Prefer before 2018 TPAC
  - [x] I have read and filled out the [Self-Review Questionnare on Security and Privacy](https://www.w3.org/TR/security-privacy-questionnaire/). The [assessment is here](https://gist.github.com/binji/c442444e222d7a0d060cba0afb130e1a).
  - [x] I have reviewed the TAG's [API Design Principles](https://w3ctag.github.io/design-principles/)

You should also know that...

[please tell us anything you think is relevant to this review]

We'd prefer the TAG provide feedback as (please select one):

  - [x] open issues in our Github repo for each point of feedback
  - [ ] open a single issue in our Github repo for the entire review
  - [ ] leave review feedback as a comment in this issue and @-notify [github usernames]

**Comment by travisleithead** [comment]
The TAG reviewed this today and while we're not deep experts in Atomics, we appreciated the well-written explainer that helped us get a pretty good idea of what was being proposed, how it might work, and what new primitives are being introduced. Thank you so much for the explainer--we can't stress enough how much we love seeing a well-written explainer (spread the word!). We also noted the Security and Privacy review and agree with your assessments. We appreciated the use of Workers and (generally) leveraging the existing platform to build-up the WASM threading model. Thanks for reviewing with us!
